packages/next/src/build/create-compiler-aliases.ts

```diff
@@ -259,9 +259,9 @@ export function createRSCAliases(
     'react-dom/static$': `next/dist/compiled/react-dom-experimental/static`,
     'react-dom/static.edge$': `next/dist/compiled/react-dom-experimental/static.edge`,
     'react-dom/static.browser$': `next/dist/compiled/react-dom-experimental/static.browser`,
-    // optimizations to ignore the legacy build of react-dom/server in `server.browser` build
-    'react-dom/server.edge$': `next/dist/build/webpack/alias/react-dom-server-edge${bundledReactChannel}.js`,
-    'react-dom/server.browser$': `next/dist/build/webpack/alias/react-dom-server-browser${bundledReactChannel}.js`,
+    // TODO: restore optimizations to ignore the legacy build of react-dom/server in `server.browser` build
+    'react-dom/server.edge$': `next/dist/compiled/react-dom${bundledReactChannel}/server.edge`,
+    'react-dom/server.browser$': `next/dist/compiled/react-dom${bundledReactChannel}/server.browser`,
```

Comment:
This reverts https://github.com/vercel/next.js/pull/56082 partially.`

However, `next/dist/compiled/next-server/app-page.runtime.prod.js` is only 5kB larger than on `canary` (330kB instead of 325kB). Doesn't seem like this change regresses anything significantly while improving correctness. @feedthejim should confirm though if there is another bundle that I should check.

Reply:
Before:
![Screenshot 2024-04-25 at 12 12 29](https://github.com/vercel/next.js/assets/12292047/c089db68-a611-4a40-b798-57851d130fca)

-- https://sebbie-vercel.vercel.app/react-18-sync-stats/report.prod-webpack-stable-app-canary-909789aaf409f075af954d58870fdf8dae0c90b5.html

After:
![Screenshot 2024-04-25 at 12 12 43](https://github.com/vercel/next.js/assets/12292047/cda7c8b9-8faf-405a-ae19-249980ee6df7)

-- https://sebbie-vercel.vercel.app/react-18-sync-stats/report.prod-webpack-stable-app.react-18-sync-fbd61741cbb86048cb690a82a5fccacbfd6c0130.html

Reply:
![Screenshot 2024-04-25 at 12 34 53](https://github.com/vercel/next.js/assets/12292047/438253b6-ac46-4b6a-a52b-4e1722cb980e)